docs/file-formats.md

```diff
@@ -0,0 +1,402 @@
+---
+jupytext:
+  text_representation:
+    extension: .md
+    format_name: myst
+    format_version: 0.12
+    jupytext_version: 1.9.1
+kernelspec:
+  display_name: Python 3
+  language: python
+  name: python3
+---
+
+:::{currentmodule} tskit
+:::
+
+(sec_file_formats)=
+
+# File formats
+
+
+(sec_tree_sequence_file_format)=
+
+## Tree sequence file format
+
+To make tree sequence data as efficient and easy as possible to use, we store the
+data on file in a columnar, binary format. The format is based on the
+[kastore](https://pypi.org/project/kastore/) package, which is a simple
+key-value store for numerical data. There is a one-to-one correspondence
+between the tables described above and the arrays stored in these files.
+
+By convention, these files are given the `.trees` suffix (although this
+is not enforced in any way), and we will sometimes refer to them as ".trees"
+files. We also refer to them as "tree sequence files".
+
+:::{todo}
+Link to the documentation for kastore, and describe the arrays that are
+stored as well as the top-level metadata.
+:::
+
+% The root group contains two attributes, `format_version` and `sequence_length`.
```

Comment:
Might as well delete this commented out stuff, it's massively out of date.

Reply:
Great. Done